### PR TITLE
Gateway segregation

### PR DESCRIPTION
**Pull request by arambaghdyan**
Separated Gateway from TourManager.API.
Introduced Ocelot as a gateway.
Added Consul as service discovery.
Moved controllers from TorManager.API to appropriate services.
Added *.Grpc projects for each service containing everything connected to grpc.
Changed Protobuf option to generate both server and client codes.
Changed namespaces.
Fixed namespaces in some projects.
Removed useless codes.
Added .env file to keep common environment configurations for docker compose.

### CODE REVIEW

File: src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+
+namespace Travely.Common.ServiceDiscovery
+{
+    public static class ServiceDiscoveryExtensions
+    {
+        public static void RegisterConsulServices(this IServiceCollection services, IConfiguration configuration)
+        {
+            services.Configure<ConsulClientConfiguration>(configuration.GetSection(nameof(ConsulClientConfiguration)));
+            services.Configure<ServiceDiscoveryConfiguration>(configuration.GetSection(nameof(ServiceDiscoveryConfiguration)));
+            services.AddSingleton<IHostedService, ServiceDiscoveryHostedService>();
```
Comment (arttonoyan) [inline src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs]: services.AddHostedService<ServiceDiscoveryHostedService>();

File: src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+
+
+namespace Travely.Common.ServiceDiscovery
+{
+    public static class ServiceDiscoveryExtensions
+    {
+        public static void RegisterConsulServices(this IServiceCollection services, IConfiguration configuration)
```
Comment (arttonoyan) [inline src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs]: I would change to Microsoft's style:
public static IServiceCollection AddConsulServices(this IServiceCollection services, IConfiguration configuration)

**Review by arttonoyan — DISMISSED** [review]

File: src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs (hunk 1 above)
Comment (arambaghdyan) [reply src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs]: Done

File: src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs (hunk 2 above)
Comment (arambaghdyan) [reply src/Commons/Travely.Common.ServiceDiscovery/ServiceDiscoveryExtensions.cs]: Done

**Pull request merged**